tests/phpunit/tests/block-supports/wpRenderElementsSupport.php

```diff
@@ -18,6 +18,134 @@ public function tear_down() {
+	 *
+	 * @covers ::wp_render_elements_support
+	 *
+	 * @param array  $color_settings  The color block support settings used for elements support.
+	 * @param array  $elements_styles The elements styles within the block attributes.
+	 * @param string $block_markup    Original block markup.
+	 * @param string $expected_markup Resulting markup after application of elements block support.
+	 *
+	 * @return void
+	 */
+	public function test_leaves_empty_block_content_empty( $color_settings, $elements_styles, $block_markup, $expected_markup ) {
+		$this->test_block_name = 'test/element-block-supports';
+
+		register_block_type(
+			$this->test_block_name,
+			array(
+				'api_version' => 3,
+				'attributes'  => array(
+					'style' => array(
+						'type' => 'object',
+					),
+				),
+				'supports'    => array(
+					'color' => $color_settings,
+				),
+			)
+		);
+
+		$block = array(
+			'blockName' => $this->test_block_name,
+			'attrs'     => array(
+				'style' => array(
+					'elements' => $elements_styles,
+				),
+			),
+		);
+
+		$actual = wp_render_elements_support( '', $block );
+
+		$this->assertSame( '', $actual, 'Expected empty block output HTML but found rendered markup instead.' );
+	}
+
+	/**
+	 * Tests that no output is generated when a block's attributes are missing.
+	 *
+	 * No block supports should be active without an attribute requesting so.
+	 *
+	 * @ticket 59578
+	 *
+	 * @dataProvider data_elements_block_support_class
+	 *
+	 * @covers ::wp_render_elements_support
+	 *
+	 * @param array  $color_settings  The color block support settings used for elements support.
+	 * @param array  $elements_styles The elements styles within the block attributes.
+	 * @param string $block_markup    Original block markup.
+	 * @param string $expected_markup Resulting markup after application of elements block support.
+	 *
+	 * @return void
+	 */
+	public function test_leaves_unrequested_block_content_as_is( $color_settings, $elements_styles, $block_markup, $expected_markup ) {
+		$this->test_block_name = 'test/element-block-supports';
+
+		register_block_type(
+			$this->test_block_name,
+			array(
+				'api_version' => 3,
+				'attributes'  => array(
+					'style' => array(
+						'type' => 'object',
+					),
+				),
+				'supports'    => array(
+					'color' => $color_settings,
+				),
+			)
+		);
+
+		$block = array(
+			'blockName' => $this->test_block_name,
+		);
+
+		$actual = wp_render_elements_support( $block_markup, $block );
+		$this->assertSame( $block_markup, $actual, 'Expected an unmodified block but found transformation.' );
+
+		$actual_when_empty = wp_render_elements_support( '', $block );
+		$this->assertSame( '', $actual_when_empty, 'Expected empty block output HTML but found content instead.' );
+	}
+
+	/**
+	 * Tests that block supports leaves block content alone if the block type isn't registered.
+	 *
+	 * @ticket 59578
+	 *
+	 * @dataProvider data_elements_block_support_class
```

Comment:
While this test's function should remain, it shouldn't have all the data sets from `data_elements_block_support_class` passed through it.